conf/machine/imx6qdlsabreauto.conf

```diff
@@ -32,18 +32,25 @@ KERNEL_DEVICETREE_use-mainline-bsp = " \
     imx6dl-sabreauto.dtb \
 "
 
-UBOOT_MACHINE ?= "mx6sabreauto_defconfig"
+### u-boot-fslc settings ###
 
-# These u-boot variables default to the correct settings for u-boot-imx.
-# Set overrides for u-boot-fslc.
-UBOOT_MAKE_TARGET_pn-u-boot-fslc = "all"
-UBOOT_SUFFIX_pn-u-boot-fslc = "img"
 SPL_BINARY_pn-u-boot-fslc = "SPL"
-WKS_FILE_pn-u-boot-fslc = "imx-uboot-spl-bootpart.wks.in"
+UBOOT_MACHINE_pn-u-boot-fslc ?= "mx6sabreauto_defconfig"
+UBOOT_SUFFIX_pn-u-boot-fslc = "img"
+
+### u-boot-imx settings ###
+
+# The u-boot-imx does not provide unified functionality for DL/Q/QP SoC
+# variants. Change the defconfig to the targeted SoC variant.
+SPL_BINARY_pn-u-boot-imx = ""
+UBOOT_MACHINE_pn-u-boot-imx = "mx6qsabreauto_defconfig"
+UBOOT_MAKE_TARGET_pn-u-boot-imx = "u-boot.imx"
+UBOOT_SUFFIX_pn-u-boot-imx = "imx"
 
-# The fsl distro uses u-boot-imx which does not provide unified functionality
-# for dl/q/qp SoC variants. Change the defconfig to the targeted SoC variant.
-UBOOT_MACHINE_fsl = "mx6qsabreauto_defconfig"
+WKS_FILE = " \
+    ${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', \
+                         'u-boot-fslc', 'imx-uboot-spl-bootpart.wks.in', \
+                                        'imx-uboot-bootpart.wks.in', d)}"
```

Comment:
Can you use `WKS_FILE_pn-` instead?

Reply:
You can't. It is used in another context and not inside the U-Boot recipe. This was the root cause of @dv1 error, in fact.

Reply:
I see, perhaps you can make a comment then so it's clear why it doesn't follow the same pattern?

Reply:
I added some other comments that try to make things clearer, but I don't see how it'd help as it is done this way all around the layer, so people would need to comment on many recipes.

Reply:
This is all I had in mind:

```
# For WKS_FILE we can't use _pn-u-boot-* override like above since WKS_FILE is used outside of u-boot context.
```

I guess that fix gets kind of buried.